include/openthread/platform/crypto.h

```diff
@@ -0,0 +1,537 @@
+ *
+ */
+otError otPlatCryptoGetKeyAttributes(otCryptoKeyRef aKeyId, otCryptoKeyAttributes *aKeyAttributes);
+
+/**
+ * Get the Crypto type supported by the platform.
+ *
+ * @retval OT_CRYPTO_TYPE_MBEDTLS If the platform supports mbedTLS.
+ * @retval OT_CRYPTO_TYPE_PSA     If platform supports PSA.
+ *
+ * @note This API is available only if platform supports PSA crypto
+ *
+ */
+otCryptoType otPlatCryptoGetType(void);
+
+/**
+ * Initialise the HMAC operation.
+ *
+ * @param[in]  aContext          Context for HMAC operation.
+ * @param[in]  aContextSize      Context size HMAC operation.
+ *
+ * @retval OT_ERROR_NONE          Successfully initialised HMAC operation.
+ * @retval OT_ERROR_FAILED        Failed to initialise HMAC operation.
+ * @retval OT_ERROR_INVALID_ARGS  @p aContext was NULL
+ *
+ * @note Incase PSA is supported pointer to psa_mac_operation_t will be passed as input.
+ * In case of mbedTLS, pointer to  mbedtls_md_context_t will be provided.
+ *
+ */
+otError otPlatCryptoHmacSha256Init(void *aContext, size_t aContextSize);
+
+/**
+ * UnInitialise the HMAC operation.
+ *
+ * @param[in]  aContext          Context for HMAC operation.
+ *
+ * @retval OT_ERROR_NONE          Successfully uninitialised HMAC operation.
+ * @retval OT_ERROR_FAILED        Failed to uninitialised HMAC operation.
+ * @retval OT_ERROR_INVALID_ARGS  @p aContext was NULL
+ *
+ * @note Incase PSA is supported pointer to psa_mac_operation_t will be passed as input.
+ * In case of mbedTLS, pointer to  mbedtls_md_context_t will be provided.
+ *
+ */
+otError otPlatCryptoHmacSha256UnInit(void *aContext, size_t aContextSize);
+
+/**
+ * Start HMAC operation.
+ *
+ * @param[in]  aContext           Context for HMAC operation.
+ * @param[in]  aContextSize       Context size HMAC operation.
+ * @param[in]  aKey               Key material to be used for for HMAC operation.
+ *
+ * @retval OT_ERROR_NONE          Successfully started HMAC operation.
+ * @retval OT_ERROR_FAILED        Failed to start HMAC operation.
+ * @retval OT_ERROR_INVALID_ARGS  @p aContext or @p aKey was NULL
+ *
+ * @note In case PSA is supported pointer to psa_mac_operation_t will be passed as input.
+ * In case of mbedTLS, pointer to  mbedtls_md_context_t will be provided.
+ *
+ */
+otError otPlatCryptoHmacSha256Start(void *aContext, size_t aContextSize, otCryptoKey *aKey);
+
+/**
+ * Update the HMAC operation with new input.
+ *
+ * @param[in]  aContext           Context for HMAC operation.
+ * @param[in]  aContextSize       Context size HMAC operation.
+ * @param[in]  aBuf               A pointer to the input buffer.
+ * @param[in]  aBufLength         The length of @p aBuf in bytes.
+ *
+ * @retval OT_ERROR_NONE          Successfully updated HMAC with new input operation.
+ * @retval OT_ERROR_FAILED        Failed to update HMAC operation.
+ * @retval OT_ERROR_INVALID_ARGS  @p aContext or @p aBuf was NULL
+ *
+ * @note Incase PSA is supported pointer to psa_mac_operation_t will be passed as input.
```

Comment:
```suggestion
 * @note In case PSA is supported pointer to psa_mac_operation_t will be passed as input.
```

nit: Same for the below docs.